sycl/include/CL/sycl/accessor.hpp

```diff
@@ -310,6 +333,28 @@ class image_accessor
       throw feature_not_supported("Images are not supported by this device.");
   }
 
+#ifdef __SYCL_DEVICE_ONLY__
+
+  sycl::vec<int, Dimensions> getCountInternal() const {
+    return __invoke_ImageQuerySize<sycl::vec<int, Dimensions>, OCLImageTy>(MImageObj);
+  }
+
+  size_t getElementSize() const {
+    int ChannelType = __invoke_ImageQueryFormat<int, OCLImageTy>(MImageObj);
+    int ChannelOrder = __invoke_ImageQueryOrder<int, OCLImageTy>(MImageObj);
+    int ElementSize = getSPIRVElementSize(ChannelType, ChannelOrder);
+    return ElementSize;
+  }
+
+
+#else
+
+  sycl::vec<int, Dimensions> getCountInternal() const {
```

Comment:
Why does this method always return a vector of all ones?

Reply:
Good catch. Marked as not implemented so far. Will fix as a separate patch.

Reply:
We don't need this version of getCountInternal() you mean? So you will remove it completely?

Reply:
getCountInternal is used in __image_array_slice__, so it need to be implemented. I believe it should return getMemoryRange() converted to vec class with appropriate dimension.